File: addon/lib/algorithm/dummy.cc

#include "dummy.h"

namespace cryptian {

namespace algorithm {

std::string Dummy::getName() {
    return "Dummy";
}

std::size_t Dummy::getVersion() {
    return 20180627;
}

std::size_t Dummy::getBlockSize() {
    return 8;
}

std::vector<std::size_t> Dummy::getKeySizes() {
    return {8};
}

std::vector<char> Dummy::encrypt(const std::vector<char> plaintext) {

	return plaintext;

}

std::vector<char> Dummy::decrypt(const std::vector<char> ciphertext) {

    return ciphertext;

}

};

};

File: test/algorithm/dummy.js

const {algorithm} = require('../..');
const assert = require('assert');

(typeof algorithm.Dummy === 'function'? describe : describe.skip)
('dummy', () => {

    const ciphertext = Buffer.from('a1502d70ba1320c8', 'hex');
    const plaintext  = Buffer.from('0001020304050607', 'hex');

    it('should encrypt', () => {

        const dummy = new algorithm.Dummy();

        assert(ciphertext.equals(dummy.encrypt(ciphertext)), 'encrypted plaintext should equal to ciphertext');

    });
    
    it('should decrypt', () => {

        const dummy = new algorithm.Dummy();

        assert(plaintext.equals(dummy.decrypt(plaintext)), 'decrypted ciphertext should equal to plaintext');

    });


});